cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSpecChecker.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.api.kafka.model.KafkaSpec;
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.api.kafka.model.status.ConditionBuilder;
+import io.strimzi.api.kafka.model.storage.Storage;
+import io.strimzi.operator.cluster.model.KafkaCluster;
+import io.strimzi.operator.cluster.model.KafkaConfiguration;
+import io.strimzi.operator.cluster.model.ZookeeperCluster;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Checks for potential problems in the configuration requested by the user, to provide
+ * warnings and share best practice. The intent is this class will generate warnings about
+ * configurations that aren't necessarily illegal or invalid, but that could potentially
+ * lead to problems.
+ */
+public class KafkaSpecChecker {
+
+    private KafkaSpec spec;
+    private KafkaCluster kafkaCluster;
+    private ZookeeperCluster zkCluster;
+    private String timestamp;
+
+    /**
+     * @param spec The spec requested by the user in the CR
+     * @param kafkaCluster The model generated based on the spec. This is requested so that default
+     *                     values not included in the spec can be taken into account, without needing
+     *                     this class to include awareness of what defaults are applied.
+     * @param zkCluster The model generated based on the spec. This is requested so that default
+     *                     values not included in the spec can be taken into account, without needing
+     *                     this class to include awareness of what defaults are applied.
+     */
+    public KafkaSpecChecker(KafkaSpec spec, KafkaCluster kafkaCluster, ZookeeperCluster zkCluster) {
+        this.spec = spec;
+        this.kafkaCluster = kafkaCluster;
+        this.zkCluster = zkCluster;
+        this.timestamp = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").format(new Date());
```

Comment:
Can we factor out a utility method for creating the timestamp (maybe put it in `ModelUtils`)? Because I know there are at least a few other places where we do this (`grep -F '"yyyy-MM-dd'T'HH:mm:ssZ"'`)

Reply:
Nice idea, thanks